sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSemaphore.scala

```diff
@@ -316,7 +319,8 @@ private final class SemaphoreTaskInfo(val stageId: Int, val taskAttemptId: Long)
     if (hasSemaphore) {
       semaphore.release(numPermits)
       hasSemaphore = false
-      lastHeld = System.currentTimeMillis()
+      lastReleased = System.nanoTime()
+      GpuTaskMetrics.get.addGpuTime(lastReleased - lastAcquired)
```

Comment:
Should we call it `semaphoreTime` instead? Because what we measure is how long the task has held the semaphore.

Reply:
semaphoreTime could also be ambiguous because it could either mean "time acquiring semaphore" or "time having the semaphore". In fact, if you take a looks at the equation: "total_core_time = total_gpu_time + total_acquire_gpu_time + other_time_spent_on_cpu", it is somehow intuitive to call it "gpuTime"?

Reply:
We can call it `semaphoreHoldingTime` if it's not clear. The reason I prefer `semaphore` to `gpu` is `semaphore` is less misleading. We may not always use the gpu for the whole time while holding the semaphore. Though, I realized that the naming cannot solve the ambiguity problem completely. I would suggest adding some comment [here](https://github.com/NVIDIA/spark-rapids/pull/11821/files#diff-27c51788980170410e3ee5189b52e14c8951c1baebd68ab39b10448a06dc6228R186) that explains exactly how this metric is measured. The naming will be less important once documentation is done.

Reply:
renamed to semaphoreHoldingTime. since semaphoreHoldingTime is quite self explainatory, I skipped the comment